src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/IDocumentSnapshot.cs

```diff
@@ -31,5 +31,7 @@ internal interface IDocumentSnapshot
     bool TryGetTextVersion(out VersionStamp result);
     bool TryGetGeneratedOutput([NotNullWhen(true)] out RazorCodeDocument? result);
 
+    Task<RazorCodeDocument> GetFormatterCodeDocumentAsync();
```

Comment:
Could you refactor to avoid adding this method directly onto the `IDocumentSnapshot` interface?

Reply:
Yes, I added more general GetGeneratedDocumentAsync that takes a parameter indicating whether designtime should be preferred, and added a separate IFormattingCodeDocumentProviderAsync service that has different implementation in Lsp vs Remote.